src/coprocessor/dag/executor/index_scan.rs

```diff
@@ -123,21 +112,39 @@ impl<'a> IndexScanExecutor<'a> {
         }
         Ok(Some(Row::new(handle, values)))
     }
+
+    fn new_scanner(&self, range: KeyRange) -> Result<Scanner> {
+        Scanner::new(&self.store, ScanOn::Index, self.desc, false, range).map_err(Error::from)
+    }
 }
 
-impl<'a> Executor for IndexScanExecutor<'a> {
+impl Executor for IndexScanExecutor {
     fn next(&mut self) -> Result<Option<Row>> {
-        while self.cursor < self.key_ranges.len() {
-            let data = self.get_row_from_range()?;
-            if data.is_none() {
+        loop {
+            if let Some(row) = self.get_row_from_range_sanner()? {
                 CORP_GET_OR_SCAN_COUNT.with_label_values(&["range"]).inc();
```

Comment:
Could we modify`CORP_GET_OR_SCAN_COUNT` in `scanner.next_row()`, so we do not need to modify it in both index_scan && table_scan?